From 6c1113633fde51b0e60f02243cfad1b3d09762cc Mon Sep 17 00:00:00 2001
From: Akhilesh Nema <nemaakhilesh@gmail.com>
Date: Tue, 2 Dec 2025 18:11:24 -0800
Subject: [PATCH] fix 'UINT_MAX' undeclared with build with musl libc

- utils_math.c:136:20: error: 'UINT_MAX' undeclared (first use in this function)
- tc_core.c:51:22: error: 'UINT_MAX' undeclared (first use in this function)

Signed-off-by: Akhilesh Nema <nemaakhilesh@gmail.com>
---
 lib/utils_math.c | 1 +
 tc/tc_core.c     | 1 +
 2 files changed, 2 insertions(+)

diff --git a/lib/utils_math.c b/lib/utils_math.c
index a7e74744..fd2ddc7c 100644
--- a/lib/utils_math.c
+++ b/lib/utils_math.c
@@ -4,6 +4,7 @@
 #include <stdlib.h>
 #include <string.h>
 #include <math.h>
+#include <limits.h>
 #include <asm/types.h>
 
 #include "utils.h"
diff --git a/tc/tc_core.c b/tc/tc_core.c
index a422e02c..b13b7d78 100644
--- a/tc/tc_core.c
+++ b/tc/tc_core.c
@@ -11,6 +11,7 @@
 #include <unistd.h>
 #include <fcntl.h>
 #include <math.h>
+#include <limits.h>
 #include <sys/socket.h>
 #include <netinet/in.h>
 #include <arpa/inet.h>
-- 
2.47.3

